woocommerce/woocommerce
# Add marketplace theme discover section

### Changes proposed in this Pull Request:

This PR is part of https://github.com/woocommerce/woocommerce/pull/40159

This PR adds support for theme sections on the Discover tab.
It also displays the new section on no-results for search theme.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Prep

To test the request from your local WooCommerce core wp-env environment to your local WCCOM environment, you first need a running instance of WCCOM. 

Make sure you have the latest version of trunk in your WCCOM environment. Set up a group of three or four theme products in the [Featured Products](https://woocommerce.test/wp-admin/edit.php?post_type=featured_product) ad admin. (See the test steps in 18244-gh-Automattic/woocommerce.com for more details.) Check that the group is being returned from the featured endpoint https://woocommerce.test/wp-json/wccom-extensions/2.0/featured?country=US. (You'll probably need to flush the transient that caches this on your WCCOM with `wp cache flush`.)

Add a mu-plugin to wp-env (as suggested in p6JqRr-aWb), and put these two filters in it:

```php
add_filter( 'http_request_host_is_external', '__return_true' );

add_filter( 'http_request_args', function ( $r ) {
	$r['reject_unsafe_urls'] = false;
	$r['sslverify']          = false;
	$r['timeout']            = 60;

	return $r;
} );
```

Get the ID of wp-env's WordPress container.

<img width="500" alt="image" src="https://github.com/woocommerce/woocommerce/assets/1647564/13509f82-962c-4f88-92ce-5f39222f2f64">

Link the container to woocommerce.test in the Docker network.

```bash
docker network connect woocommerce.test <container ID>
```

#### Test

1. **Until we get WCCOM API support, we need to point the API to local WCCOM with data available in the API**
2. [Replace this line with](https://github.com/woocommerce/woocommerce/blob/HEAD/plugins/woocommerce/includes/admin/class-wc-admin-addons.php#L83):
```
$featured = false;
```
3. Replace[ these lines with](https://github.com/woocommerce/woocommerce/blob/HEAD/plugins/woocommerce/includes/admin/class-wc-admin-addons.php#L100-L101):
```
$raw_featured = wp_safe_remote_get(
				'https://woocommerce.test/wp-json/wccom-extensions/2.0/featured' . $parameter_string . '&no_cache=true',
```
4. Open [the Discover page](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions)
5. Confirm that the _Our favorite ecommerce themes._ section is loaded
6. Cards in this section should be displayed as themes



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Add support for themes to marketplace discovery and use it in the theme no-result section. 

</details>

![Screenshot 2023-09-25 at 13 31 29](https://github.com/woocommerce/woocommerce/assets/1199991/5f269c75-b7cd-4076-9e77-2118f4cc7f1d)
